ip_country/src/country_block_stream.rs

```diff
@@ -0,0 +1,411 @@
+            Ipv4Addr::from_str("4.3.2.1").unwrap(),
+        );
+
+        let result = subject.contains(IpAddr::from_str("1:2:3:4:0:0:0:0").unwrap());
+
+        assert_eq!(result, false);
+    }
+
+    #[test]
+    fn ip_range_doesnt_find_ipv4_address_in_ipv6_range() {
+        let subject = IpRange::V6(
+            Ipv6Addr::from_str("0:0:0:0:1:2:3:4").unwrap(),
+            Ipv6Addr::from_str("0:0:0:0:4:3:2:1").unwrap(),
+        );
+
+        let result = subject.contains(IpAddr::from_str("1.2.3.4").unwrap());
+
+        assert_eq!(result, false);
+    }
+
+    #[test]
+    fn try_from_works_for_ipv4() {
+        let string_record = StringRecord::from(vec!["1.2.3.4", "5.6.7.8", "AS"]);
+
+        let result = CountryBlock::try_from(string_record);
+
+        assert_eq!(
+            result,
+            Ok(CountryBlock {
+                ip_range: IpRange::V4(
+                    Ipv4Addr::from_str("1.2.3.4").unwrap(),
+                    Ipv4Addr::from_str("5.6.7.8").unwrap()
+                ),
+                country: Country::try_from("AS").unwrap().clone(),
+            })
+        );
+    }
+
+    #[test]
+    fn try_from_works_for_ipv6() {
+        let string_record = StringRecord::from(vec![
+            "1234:2345:3456:4567:5678:6789:789A:89AB",
+            "4321:5432:6543:7654:8765:9876:A987:BA98",
+            "VN",
+        ]);
+
+        let result = CountryBlock::try_from(string_record);
+
+        assert_eq!(
+            result,
+            Ok(CountryBlock {
+                ip_range: IpRange::V6(
+                    Ipv6Addr::from_str("1234:2345:3456:4567:5678:6789:789A:89AB").unwrap(),
+                    Ipv6Addr::from_str("4321:5432:6543:7654:8765:9876:A987:BA98").unwrap()
+                ),
+                country: Country::try_from("VN").unwrap().clone(),
+            })
+        );
+    }
+
+    #[test]
+    fn try_from_fails_for_bad_ip_syntax() {
+        let string_record = StringRecord::from(vec!["Ooga", "Booga", "AS"]);
+
+        let result = CountryBlock::try_from(string_record);
+
+        assert_eq!(
+            result,
+            Err("Invalid (AddrParseError(Ip)) IP address in CSV record: 'Ooga'".to_string())
+        );
+    }
+
+    #[test]
+    fn try_from_fails_for_missing_start_ip() {
+        let strings: Vec<&str> = vec![];
+        let string_record = StringRecord::from(strings);
+
+        let result = CountryBlock::try_from(string_record);
+
+        assert_eq!(result, Err("Missing IP address in CSV record".to_string()));
+    }
+
+    #[test]
+    fn try_from_fails_for_missing_end_ip() {
+        let string_record = StringRecord::from(vec!["1.2.3.4"]);
+
+        let result = CountryBlock::try_from(string_record);
+
+        assert_eq!(result, Err("Missing IP address in CSV record".to_string()));
+    }
+
+    #[test]
+    fn try_from_fails_for_reversed_ipv4_addresses() {
+        let string_record = StringRecord::from(vec!["4.3.2.1", "1.2.3.4", "ZZ"]);
```

Comment:
Hmmm. I'd rather see values as close as possible to the right ones and still having this test pass. 

```Rust
vec!["1.2.3.4", "1.2.3.3", "ZZ"]
```